### PR TITLE
expose the consumer labels on the API

### PR DESCRIPTION
**Pull request by geoberle**
this PR enables the `labels` column of the `consumers` table to be exposed on the consumer API.

these labels will allow for more usecase specific metadata on the consumer inventory, e.g. attaching identify

### CODE REVIEW

File: pkg/handlers/consumer.go [hunk 1]
```diff
@@ -61,8 +63,12 @@ func (h consumerHandler) Patch(w http.ResponseWriter, r *http.Request) {
...
 				return nil, err
 			}
-
-			//patch a field
+			if patch.Labels != nil {
+				found.Labels = db.EmptyMapToNilStringMap(patch.Labels)
+			}
+			if patch.Name != nil {
+				found.Name = util.EmptyStringToNil(*patch.Name)
+			}
```
Comment (clyang82) [inline pkg/handlers/consumer.go:71]: What is use case to change the consumer name here?

**Review by morvencao — COMMENTED** [review]
also need DB migration for this change?

File: pkg/handlers/consumer.go (hunk 1 above)
Comment (geoberle) [reply pkg/handlers/consumer.go:71]: it is mentioned as a valid patch field in the `ConsumerPatchRequest` - https://github.com/openshift-online/maestro/blob/e515b02da7105a46720a3924c21215b4a2894fb1/openapi/openapi.yaml#L554

i'm not sure the name should be changable, but if not, the field should be removed from the `ConsumerPatchRequest`openapi type.

**Comment by geoberle** [comment]
> also need DB migration for this change?

afaiu not. the labels column is already present in the database and the restriction to allow only string values of the REST API label JSON payload does not reflect on the database column type.

File: pkg/handlers/consumer.go (hunk 1 above)
Comment (clyang82) [reply pkg/handlers/consumer.go:71]: I do not know where to use consumer name. @markturansky Do you have use case to use consumer name?

Comment (clyang82) [reply pkg/handlers/consumer.go:71]: @geoberle Can you help to remove the name field from `ConsumerPatchRequest` openapi type? Thanks.

Comment (geoberle) [reply pkg/handlers/consumer.go:71]: i completely missed this comment. i will work on it now

Comment (geoberle) [reply pkg/handlers/consumer.go]: @clyang82 @morvencao can you allow the PR check to run?

Comment (markturansky) [reply pkg/handlers/consumer.go]: Consumer name isn't required, as Maestro uses the ID and expects clients to post the consumer ID

**Comment by markturansky** [comment]
@geoberle there are no labels on the consumer table.  this requires a database migration.

what labels will be provided? are they to be searchable?

**Comment by clyang82** [comment]
@geoberle I missed your update. It is ready to go when you rebase your PR and append the output of `make test; make test-integration`.

**Comment by geoberle** [comment]
> @geoberle I missed your update. It is ready to go when you rebase your PR and append the output of `make test; make test-integration`.

```
make test; make test-integration
CGO_ENABLED=1 GOEXPERIMENT=boringcrypto go install -ldflags="" ./cmd/maestro
OCM_ENV=testing gotestsum --jsonfile-timing-events=/Users/goberlec/dev/geoberle/maestro/unit-test-results.json --format short-verbose -- -p 1 -v  \
	./pkg/... \
	./cmd/...
PASS pkg/api.TestJSONMapStausToResourceStatus/empty (0.00s)
PASS pkg/api.TestJSONMapStausToResourceStatus/content_status (0.00s)
PASS pkg/api.TestJSONMapStausToResourceStatus/reconcile_status (0.00s)
PASS pkg/api.TestJSONMapStausToResourceStatus/reconcile_status_with_deleted_Condition (0.00s)
PASS pkg/api.TestJSONMapStausToResourceStatus (0.00s)
PASS pkg/api.TestResourceStatusToJSONMap/empty (0.00s)
PASS pkg/api.TestResourceStatusToJSONMap/content_status (0.00s)
PASS pkg/api.TestResourceStatusToJSONMap/reconcile_status (0.00s)
PASS pkg/api.TestResourceStatusToJSONMap/reconcile_status_with_deleted_Condition (0.00s)
PASS pkg/api.TestResourceStatusToJSONMap (0.00s)
PASS pkg/api (cached)
EMPTY pkg/api/openapi
EMPTY pkg/api/presenters
EMPTY pkg/auth
EMPTY pkg/client/cloudevents
EMPTY pkg/client/ocm
PASS pkg/config.TestConfigReadStringFile (0.00s)
PASS pkg/config.TestConfigReadIntFile (0.00s)
PASS pkg/config.TestConfigReadBoolFile (0.00s)
PASS pkg/config.TestConfigReadQuotedFile (0.00s)
PASS pkg/config (cached)
PASS pkg/controllers.TestControllerFramework (0.00s)
PASS pkg/controllers (cached)
EMPTY pkg/dao
EMPTY pkg/dao/mocks
EMPTY pkg/db
EMPTY pkg/db/db_context
EMPTY pkg/db/db_session
EMPTY pkg/db/migrations
EMPTY pkg/db/mocks
EMPTY pkg/db/transaction
EMPTY pkg/dispatcher
PASS pkg/errors.TestErrorFormatting (0.00s)
PASS pkg/errors.TestErrorFind (0.00s)
PASS pkg/errors (cached)
EMPTY pkg/event
EMPTY pkg/handlers (cached)
EMPTY pkg/logger
PASS pkg/services.TestSQLTranslation (0.02s)
PASS pkg/services.TestResourceFindByConsumerID (0.00s)
PASS pkg/services.TestCreateInvalidResource (0.00s)
PASS pkg/services.TestValidateNewManifest/validated (0.00s)
PASS pkg/services.TestValidateNewManifest/no_apiVersion (0.00s)
PASS pkg/services.TestValidateNewManifest/no_kind (0.00s)
PASS pkg/services.TestValidateNewManifest/no_name (0.00s)
PASS pkg/services.TestValidateNewManifest/has_generate_name (0.00s)
PASS pkg/services.TestValidateNewManifest/has_resource_version (0.00s)
PASS pkg/services.TestValidateNewManifest/has_deletion_grace_period_seconds (0.00s)
PASS pkg/services.TestValidateNewManifest/wrong_namespace (0.00s)
PASS pkg/services.TestValidateNewManifest/wrong_api_version_(no_version) (0.00s)
PASS pkg/services.TestValidateNewManifest/wrong_api_version_(no_version)#01 (0.00s)
PASS pkg/services.TestValidateNewManifest/wrong_api_version (0.00s)
PASS pkg/services.TestValidateNewManifest (0.00s)
PASS pkg/services.TestValidateUpdateManifest/validated (0.00s)
PASS pkg/services.TestValidateUpdateManifest/apiVersion_mismatch (0.00s)
PASS pkg/services.TestValidateUpdateManifest/kind_mismatch (0.00s)
PASS pkg/services.TestValidateUpdateManifest/name_mismatch (0.00s)
PASS pkg/services.TestValidateUpdateManifest/namespace_mismatch (0.00s)
PASS pkg/services.TestValidateUpdateManifest (0.00s)
PASS pkg/services (cached)
EMPTY pkg/util
EMPTY cmd/maestro
EMPTY cmd/maestro/agent
PASS cmd/maestro/environments.TestLoadServices (0.22s)
PASS cmd/maestro/environments (cached)
EMPTY cmd/maestro/migrate
EMPTY cmd/maestro/servecmd
EMPTY cmd/maestro/server
EMPTY cmd/maestro/server/logging

DONE 39 tests in 4.741s
CGO_ENABLED=1 GOEXPERIMENT=boringcrypto go install -ldflags="" ./cmd/maestro
OCM_ENV=testing gotestsum --jsonfile-timing-events=/Users/goberlec/dev/geoberle/maestro/integration-test-results.json --format short-verbose -- -p 1 -ldflags -s -v -timeout 1h  \
		./test/integration
I0319 11:38:49.354799   65511 framework.go:76] Initializing testing environment
I0319 11:38:49.500297   65511 framework.go:161] Using Mock OCM Authz Client
I0319 11:38:49.502602   65511 framework.go:199] Disabling Sentry error reporting
I0319 11:38:49.535686   65511 grpc_server.go:68] Serving gRPC service without TLS at 8090
I0319 11:38:49.535896   65511 healthcheck_server.go:56] Serving HealthCheck without TLS at 8083
PASS test/integration.TestConsumerGet (0.21s)
PASS test/integration.TestConsumerPost (0.17s)
PASS test/integration.TestConsumerPatch (0.29s)
PASS test/integration.TestConsumerPaging (0.31s)
PASS test/integration.TestControllerRacing (1.65s)
PASS test/integration.TestControllerReconcile (1.17s)
PASS test/integration.TestControllerSync (1.21s)
PASS test/integration.TestPulseServer (5.18s)
PASS test/integration.TestResourceGet (0.17s)
PASS test/integration.TestResourcePost (5.34s)
PASS test/integration.TestResourcePatch (0.35s)
PASS test/integration.TestResourcePaging (0.57s)
PASS test/integration.TestResourceListSearch (0.33s)
PASS test/integration.TestUpdateResourceWithRacingRequests (0.41s)
PASS test/integration.TestResourceFromGRPC (5.27s)
E0319 11:39:12.203341   65511 logger.go:129]   Unable to get all maestro instances: pq: relation "server_instances" does not exist
I0319 11:39:12.208736   65511 api_server.go:157] Web server terminated
PASS test/integration (cached)

DONE 15 tests in 2.496s
```

should i be worried about this message?
```
E0319 11:39:12.203341   65511 logger.go:129]   Unable to get all maestro instances: pq: relation "server_instances" does not exist
´´´

**Review by clyang82 — APPROVED** [review]
LGTM

**Comment by clyang82** [comment]
You do not need to worry about that error. I am working on a PR to fix that issue. Thanks @geoberle

**Pull request merged**